lib/charms/finos_legend_libs/v0/legend_operator_base.py

```diff
@@ -778,8 +782,9 @@ def _on_db_relation_broken(
     def _on_legend_gitlab_relation_joined(
             self, event: charm.RelationJoinedEvent) -> None:
         redirect_uris = self._get_legend_gitlab_redirect_uris()
-        legend_gitlab.set_legend_gitlab_redirect_uris_in_relation_data(
-            event.relation.data[self.app], redirect_uris)
+        if self.unit.is_leader():
+            legend_gitlab.set_legend_gitlab_redirect_uris_in_relation_data(
+                event.relation.data[self.app], redirect_uris)
```

Comment:
This code is pretty much the same as in ``_update_gitlab_relation_callback_uris``. We could just call that instead.